content/docs/rfcs/44/README.md

```diff
@@ -0,0 +1,261 @@
+This fee is locked in the pool and is non refundable.
+
+Querying nodes/ service consumers who wish to pay for their received services, deposit some fund in the Waku Service Contract and obtain Service Credentials or Tokens.
+Tokens have arbitrary values which is decided by the service consumer.
+
+Upon receiving a history query, the service provider measures the cost of the service e.g., in terms of the size of the history response.
+It then communicates with the querying node 1) the cost in Ether, 2) a random `request_ID` 3) the necessary information for the payment e.g., its registered Ethereum address.
+The querying node sends a [transaction](#interaction-with-the-waku-service-contract), with the `request_ID` attached to it, to the contract to spend one of its service credential to pay to the service node.
+The contract checks the validity of the payment e.g., whether the spent credential has enough fund, whether it is spent before, and whether the designated service node is registered in the contract. 
+If all the checks go through, the fund is released to the service node and an event is emitted.  
+A small portion of the released fund is considered as the service fee and is locked in the Waku service contract. 
+The service provider listens to the contract and as soon as it gets the event of the payment for that `request_ID`, it resolves the query as normal. 
+The querying node may choose to use a proxy relayer to send its transaction to the contract in order to hide its Ethereum account address hence enjoying more privacy.
+
+
+
+
+
+# Adversarial Model
+
+We consider an honest setting where both service requester and provider follow the protocol specifications. 
+The communication channel between the involved parties i.e., the querying node, the relayer (if any) and the service provider is authenticated and private.
+
+# Wire Specification
+We extend the `HistoryRPC` protobuf message with two fields of `sttlement_request` and `settlement_response` which encapsulate information related to the settlement.
+
+## Payloads
+
+```diff
+syntax = "proto3";
+
+message HistoryQuery {
+  // the first field is reserved for future use
+  string pubsubtopic = 2;
+  repeated ContentFilter contentFilters = 3;
+  PagingInfo pagingInfo = 4;
+}
+message HistoryResponse {
+  // the first field is reserved for future use
+  repeated WakuMessage messages = 2;
+  PagingInfo pagingInfo = 3;
+  enum Error {
+    NONE = 0;
+    INVALID_CURSOR = 1;
+  }
+  Error error = 4;
+}
+
++ message SttlementResponse {
+  uint64 req_size = 1;
+  uint64 cost = 2;
+  string request_id = 3;
+  uint64 history_retrieved = 4;
+}
+
++ message SttlementRequest {
+  bool with_payment = 1;
+}
+
+message HistoryRPC {
+  string request_id = 1;
+  HistoryQuery query = 2;
+  HistoryResponse response = 3;
++ SttlementRequest sttlement_request = 19;
```

Comment:
Settlement typo in a few places ("Stt" / "stt")

Reply:
fixed now